src/main/scala/com/ubirch/webui/models/keycloak/util/BareKeycloakUtil.scala

```diff
@@ -541,6 +555,21 @@ case class MemberResourceRepresentation(resource: UserResource, representation:
     }
   }
 
+  /**
+  * Attempt to make the keycloak member leave the groups contained in the groupsToLeave list.
+    * If a group is not found in the list of the keycloak user's group, then nothing happens for this particular group
+    * @param groupsToLeave List of the group name that the device will attempt to quit.
+    * @param maybeAllGroups If provided, will use this list of groups instead of querying for fresh ones
+    *                       (can be used to reduce the amount of queries against the backend).
+    */
+  private def leaveSpecifiedGroups(groupsToLeave: List[String], maybeAllGroups: Option[List[GroupRepresentation]] = None): Unit = {
+    resource.getAllGroups(maybeAllGroups) foreach { group =>
+      if (groupsToLeave.map(_.toLowerCase()).contains(group.getName.toLowerCase())) {
```

Comment:
Wouldn't it make sense maybe to always store the group names lowercase? Or always only accept one way of writing?

Reply:
That would've been wise.. Now it's too late tho :(